base/ltcmd.dtx

```diff
@@ -3996,19 +4008,15 @@
           }
       }
     \use:n
-      {
-        \tl_trim_spaces_apply:nN {#1} \@@_arg_to_keyvalue_auxi:nnn
-          {#2} {#1}
-      }
+      { \@@_arg_to_keyvalue_auxi:nnn {#1} {#2} {#3} }
   }
 \cs_new_protected:Npn \@@_arg_to_keyvalue_auxi:nnn #1#2#3
   {
     \tl_if_head_is_N_type:nTF {#1}
       { \@@_arg_to_keyvalue_auxii:nnNw {#2} {#3} #1 \q_@@_stop }
       { \@@_arg_to_keyvalue_auxv:nn {#2} {#3} }
   }
-\cs_new_protected:Npn \@@_arg_to_keyvalue_auxii:nnNw
-  #1#2#3#4 \q_@@_stop
+\cs_new_protected:Npn \@@_arg_to_keyvalue_auxii:nnNw #1#2#3#4 \q_@@_stop
   {
     \str_if_eq:nnTF {#3} { = }
```

Comment:
Though slower this should probably use something to split of the first token without imposing restrictions to the argument. I suggest the following code (if `\expanded` is required something equally fast can be built that also works for `n`-type heads if necessary):

```
\cs_new:Npn \@@_split_N_head_apply:Nn #1#2
  { \exp:w \if_false: { \fi: \@@_split_N_head_apply_aux:NNw #1#2 } }
\cs_new:Npn \@@_split_N_head_apply_aux:NNw #1#2
  {
    \exp_after:wN \exp_end:
    \exp_after:wN #1 \exp_after:wN #2 \exp_after:wN { \if_false: } \fi:
  }
```

Then change the definition of `\@@_arg_to_keyvalue_auxi:nnn` to `\@@_arg_to_keyvalue_auxiv:nnNw`:

```
\cs_new:Npn \@@_arg_to_keyvalue_auxi:nnn #1
  {
    \tl_if_head_is_N_type:nTF {#1}
      { \@@_split_N_head_apply:Nn \@@_arg_to_keyvalue_auxii:Nnnn {#1} }
      { \@@_arg_to_keyvalue_auxv:nn }
  }
\cs_new:Npn \@@_arg_to_keyvalue_auxii:Nnnn #1#2
  {
    \str_if_eq:eeTF { \exp_not:n {#1} } { = }
      { \tl_trim_spaces_apply:nN {#2} \@@_arg_to_keyvalue_auxiii:nnn }
      { \@@_arg_to_keyvalue_auxv:nn }
  }
\cs_new:Npn \@@_arg_to_keyvalue_auxiii:nnn #1
  {
    \tl_if_head_is_N_type:nTF {#1}
      { \@@_split_N_head_apply:Nn \@@_arg_to_keyvalue_auxiv:Nnnn {#1} }
      { \@@_arg_to_keyvalue_auxv:nn }
  }
\cs_new:Npn \@@_arg_to_keyvalue_auxiv:Nnnn #1#2
  {
    \str_if_eq:eeTF { \exp_not:n {#1} } { , }
      { \tl_set:Nn \ProcessedArgument {#2} \use_none:nn }
      { \@@_arg_to_keyvalue_auxv:nn }
  }
```

Reply:
I don't follow the need here: we have a standard `\tl_if_head_is_N_type:nTF` then split the `N`-type token safely. What case are you thinking fails?

Reply:
`#4` might lose braces by the `w`-grabbing, `\q_@@_stop` can't be part of the argument.

Reply:
(the latter is fine due to the private quarks rule)